test/integration/tsconfig-override/index.test.ts

```diff
@@ -0,0 +1,35 @@
+import * as fsp from 'fs/promises'
+import { join } from 'path'
+import { createIntegrationTest } from '../utils'
+
+describe('integration tsconfig-override', () => {
+  it('should use esnext output in build without override', async () => {
+    await createIntegrationTest(
+      {
+        directory: __dirname,
+      },
+      async ({ dir }) => {
+        const content = await fsp.readFile(
+          join(dir, './dist/index.js'),
+          'utf-8',
+        )
+        expect(content).toContain("var index = (()=>'index');")
+      },
+    )
+  })
+  it('should use es5 output in build', async () => {
+    await createIntegrationTest(
+      {
+        directory: __dirname,
+        args: ['--tsconfig', 'tsconfig.build.json'],
+      },
+      async ({ dir }) => {
+        const content = await fsp.readFile(
+          join(dir, './dist/index.js'),
+          'utf-8',
+        )
+        expect(content).toContain('export { index as default };')
```

Comment:
the checking result is reversed? this should be es5

Reply:
it is using the correct tsconfig though 🤔 

![CleanShot 2024-03-14 at 00 23 51@2x](https://github.com/huozhi/bunchee/assets/51714798/77a052e3-3ccc-40dd-a042-bc013c23b55d)

Reply:
I see 👍
I changed to class, it looks easier to understand between `function` and `class` now